libraries/apollo-normalized-cache-api-incubating/build.gradle.kts

```diff
@@ -15,6 +15,9 @@ kotlin {
         api(project(":apollo-mpp-utils"))
         implementation(libs.okio)
         api(libs.uuid)
+        implementation(libs.atomicfu.get().toString()) {
+          because("Use of ReentrantLock for Apple (we don't use the gradle plugin rewrite)")
```

Comment:
Might be worth actually using the plugin to remove our own `com.apollographql.apollo3.cache.normalized.api.internal.Lock` and make lock a no-op on JS (and probably save a few instructions on JVM/Native)?

Reply:
Actually the plugin is only useful for atomic, which we don't use (yet!), whereas we need locks to ensure consistency between the caches in the chain.